src/utils.ts

```diff
@@ -12,17 +13,30 @@ export namespace Utils {
         const esc = encodeURIComponent
         const headers = { 'authorization': payload['CF_API_KEY']! }
         const runtimeName = payload['CF_RUNTIME_NAME']
+        const platformHost = payload['CF_PLATFORM_URL']
+        const runtimeVersion = await Utils.getRuntimeVersion(headers, runtimeName, platformHost)
+        const shouldStorePayloadInHeaders = gte(runtimeVersion, '0.0.553')
         let host
         if (!runtimeName) {
             host = payload['CF_HOST']
             delete payload['CF_HOST']
         } else {
-            const platformHost = payload['CF_PLATFORM_URL']
             host = await Utils.getRuntimeIngressHost(runtimeName, headers, platformHost)
             delete payload['CF_RUNTIME_NAME']
             delete payload['CF_PLATFORM_URL']
         }
         delete payload['CF_API_KEY']
+        if (shouldStorePayloadInHeaders) {
+            const dockerFileContent = payload['CF_DOCKERFILE_CONTENT']
+            if (dockerFileContent) {
+                headers['X-CF-DOCKERFILE-CONTENT'] = dockerFileContent
+                delete payload['X-CF-DOCKERFILE-CONTENT']
+            }
+            const qs = Object.entries(payload).map(kv => `${kv[0]}=${kv[1] || ''}`).join('&')
```

Comment:
It's not a real query string since escaping is missing, let's rename to `qsNoEscaping` and exaplain that this way we are tring to reduce the size of the message.